WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Grouped Products/GroupedProductListSelectorDataSource.swift

```diff
@@ -0,0 +1,112 @@
+import UIKit
+import Yosemite
+
+/// Configures the results and cells for a paginated list of linked products of a grouped product, syncs each page of products,
+/// and handles actions that could alter the linked products.
+final class GroupedProductListSelectorDataSource: PaginatedListSelectorDataSource {
+    typealias StorageModel = StorageProduct
+
+    lazy var customResultsSortOrder: ((Product, Product) -> Bool)? = { [weak self] (lhs, rhs) in
+        guard let self = self else {
+            return true
+        }
+        let lhsProductID = lhs.productID
+        let rhsProductID = rhs.productID
+        let productIDs = self.groupedProductIDs
+        guard let lhsProductIDIndex = productIDs.firstIndex(of: lhsProductID), let rhsProductIDIndex = productIDs.firstIndex(of: rhsProductID) else {
+            return true
+        }
+        return lhsProductIDIndex < rhsProductIDIndex
+    }
+
+    // Observable list of the latest grouped product IDs
+    var productIDs: Observable<[Int64]> {
+        productIDsSubject
+    }
+    private let productIDsSubject: PublishSubject<[Int64]> = PublishSubject<[Int64]>()
+
+    private(set) var groupedProductIDs: [Int64] = [] {
+        didSet {
+            if groupedProductIDs != oldValue {
+                productIDsSubject.send(groupedProductIDs)
+            }
+        }
+    }
+
+    var selected: Product?
+
+    private let siteID: Int64
+    private let product: Product
+    private let imageService: ImageService
+
+    init(product: Product, imageService: ImageService = ServiceLocator.imageService) {
+        self.siteID = product.siteID
+        self.product = product
+        self.groupedProductIDs = product.groupedProducts
+        self.imageService = imageService
+    }
+
+    func createResultsController() -> ResultsController<StorageProduct> {
+        let storageManager = ServiceLocator.storageManager
+        let predicate = NSPredicate(format: "siteID == %lld AND productID IN %@", siteID, groupedProductIDs)
+        return ResultsController<StorageProduct>(storageManager: storageManager,
+                                                 matching: predicate,
+                                                 sortOrder: .nameAscending)
+    }
+
+    func handleSelectedChange(selected: Product) {
+        // no-op
```

Comment:
Is it a `TODO` case, or we don't need in the future to handle the selected product?
It seems that need to be `self.selected = selected` since the variable `selected` is not used at the moment.

Reply:
in this use case, products are not selectable and the only UX is the delete button on product rows and the CTA to add more products (separate screens). since Swift doesn't support optional functions/variables and this data source conforms to `PaginatedListSelectorDataSource`, `func handleSelectedChange(selected:)` and `selected` are not used but have to be present 😐 I added more comments in https://github.com/woocommerce/woocommerce-ios/pull/2505/commits/e6971eac90a2ab9071a2a978bf80e1390e7c5d86